source/extensions/filters/http/compressor/compressor_filter.cc

```diff
@@ -198,7 +200,11 @@ Http::FilterHeadersStatus CompressorFilter::decodeHeaders(Http::RequestHeaderMap
   }
 
   const auto& response_config = config_->responseDirectionConfig();
-  if (compressionEnabled(response_config) && response_config.removeAcceptEncodingHeader()) {
+  const auto* per_route_config =
```

Comment:
Now that it's needed in multiple places, it would be best to make per_route_config a member variable (initialized in decodeHeaders). Having done that, it might be more readable to have compressionEnabled and removeAcceptEncodeHeader directly use config_->responseDirectionConfig() and per_route_config_, than to pass them as args.

Reply:
It's okay to keep the raw pointer around as a member?
https://github.com/envoyproxy/envoy/blob/006a20dbd3789339ee8487e304d8f70b61de5508/source/common/http/utility.h#L631-L632

Reply:
hmm... i guess it might be possible for something to change the route between decodeHeaders and encodeHeaders, and it sounds like that would invalidate the pointer. might be best to leave it as-is (because the current form is definitely safe). if anyone has a better understanding of this, feel free to chime in

Reply:
I've seen bugs sneak in due to route changes etc between decode and encode invalidating pointers to headers, etc. so I'd lean towards the resolving as we're doing.

Reply:
I will do nothing then.